sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/Uri.java

```diff
@@ -183,7 +184,25 @@ public boolean shouldRefreshHealthStatus() {
     }
 
     public String getHealthStatusDiagnosticString() {
-        return this.healthStatusTuple.get().diagnsoticString;
+        return this.healthStatusTuple.get().getDiagnsoticString();
+    }
+
+    public void setHealthStatusTupleAttempting(boolean attempting) {
+        this.healthStatusTuple.updateAndGet(previousStatusTuple ->
+            {
+                previousStatusTuple.setAttempting(attempting);
+                return previousStatusTuple;
+            }
+        );
+    }
+
+    public void setHealthStatusTuplePrimary(boolean primary) {
```

Comment:
the uri will be the full physical addresses -> including partition and replica etc. It will stay as primary as it is until gateway return a different story, why we need to dynamically set the primary flag here? 

it can be part of the diagnostics string

Reply:
removed and used the uri isPrimary variable to precalculate the diagnostic string